1-Draft/RFCNNNN-AdditionalTelemetry.md

```diff
@@ -0,0 +1,175 @@
+- What versions of PowerShell tools and services should Microsoft continue to support?
+- What PowerShell integration scenarios are people using? How many people are using PowerShell
+  Editor services
+- How can we optimize the engine size and efficiency of PowerShell for cloud scenarios?
+
+To ensure we are getting an accurate picture of how everyone uses PowerShell, not just those most
+vocal/involved in the community, we need to make improvements in our telemetry.
+PowerShell usage telemetry will allow us to better prioritize testing, support, and investments.
+It should be straightforward and easy to opt-out of PowerShell telemetry (presently the process is
+not intuitive nor well documented).
+To better share our plans and successes with the community we want more complete telemetry on our
+[Public PowerBi report](https://msit.powerbi.com/view?r=eyJrIjoiYTYyN2U3ODgtMjBlMi00MGM1LWI0ZjctMmQ3MzE2ZDNkMzIyIiwidCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsImMiOjV9&pageName=ReportSection5).
+Simply put, we want to make PowerShell better and believe this can be achieved by better
+understanding how PowerShell is being used.
+
+### Non-Goals
+
+The goal of this is **not** to collect any personal data of PowerShell users or infringe on
+users' privacy.
+Respecting the privacy of our users has been the principal factor in determining which metrics we
+plan to collect.
+The goal of this is not to collect patterns of individual users but to look at aggregate usage trends.
+
+## Specification
+
+Add telemetry to track the following metrics:
+
+- Count of unique devices (PowerShell)
+- Count of unique users (PowerShell)
+- Count of execution types (ex. cmdlets, native binaries/applications) (PowerShell)
+- Count of session types (ex. hosted vs not hosted sessions) (PowerShell)
+- Hosting application of the of the PowerShell session (ex. VSCode, ISE, AZDevOps) (PowerShell)
+- Version of PowerShell, PowerShellGet, PackageManagement, Nuget and platform
+  used to install from PSGallery (PowerShellGet)
+
+### Other features
+
+- Public PowerBi DashBoard to make the aggregated statistics available to the community.
+  Find a mocked-up PowerBi DashBoard with randomly generated data at the end of the document.
+  It includes the following reports:
+  - Unique user trends
+  - Unique devices trends
+  - Types of user &mdash;based on types of executions taking place
+  - VSCode usage trends
+  - Application trends
+  - Hosted Scenario Trends
+  - PowerShellGet installation trends
+- A cmdlet for providing feedback (Send-PSFeedback):
+  - This cmdlet would take in a string and send it back to the PowerShell team
+  - The cmdlet will also send Platform data (OS/version) and Shell data ($PSVersionTable)
+  - The cmdlet will have an -OpenGitHubIssue switch that will allow the user to open a GitHubIssue
+    in the PowerShell/PowerShell repository
+  - The user will have the option to include the the last error using an -IncludeError switch

```

Comment:
Is there value in automatically collecting a count of fully qualified error id's (which wouldn't contain any private information)?  If we know that there's a large number of CommandNotFound errors, we can invest in improving suggestions to auto-correct, etc... or if there's a bunch of other types of errors it may give us a hint of where users are having trouble.  The `-IncludeError` will provide more specific details, but is opt-in whereas I'm thinking of something automatic that's opt-out.